cmd/migration/v0.9.1-v1/v091-domain/market_domain.go

```diff
@@ -0,0 +1,277 @@
+package v091domain
+
+import (
+	"encoding/hex"
+	"errors"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/shopspring/decimal"
+)
+
+const BalancedReservesType = 1
+
+type Market struct {
+	AccountIndex        int
+	BaseAsset           string
+	QuoteAsset          string
+	BaseAssetPrecision  uint
+	QuoteAssetPrecision uint
+	Fee                 int64
+	FixedFee            FixedFee
+	Tradable            bool
+	Strategy            MakingStrategy
+	Price               Prices
+}
+
+func (m Market) AccountName() string {
+	buf, _ := hex.DecodeString(m.BaseAsset + m.QuoteAsset)
+	return hex.EncodeToString(btcutil.Hash160(buf))[:5]
+}
+
+type Prices struct {
+	BasePrice  decimal.Decimal
+	QuotePrice decimal.Decimal
+}
+
+type FixedFee struct {
+	BaseFee  int64
+	QuoteFee int64
+}
+
+type MakingStrategy struct {
+	Type    int
+	formula MakingFormula
+}
+
+type MakingFormula interface {
+	SpotPrice(spotPriceOpts interface{}) (spotPrice decimal.Decimal, err error)
+	OutGivenIn(outGivenInOpts interface{}, amountIn decimal.Decimal) (amountOut decimal.Decimal, err error)
+	InGivenOut(inGivenOutOpts interface{}, amountOut decimal.Decimal) (amountIn decimal.Decimal, err error)
+	FormulaType() int
+}
+
+func NewStrategyFromFormula(
+	formula MakingFormula,
+) MakingStrategy {
+	strategy := MakingStrategy{
+		Type:    formula.FormulaType(),
+		formula: formula,
+	}
+
+	return strategy
+}
+
+type StrategyType int32
+
+const (
+	StrategyTypePluggable StrategyType = 0
+)
+
+var (
+	ErrInvalidOptsType = errors.New("opts must be of type BalancedReservesOpts")
+	ErrAmountTooLow    = errors.New("provided amount is too low")
+	ErrAmountTooBig    = errors.New("provided amount is too big")
+	ErrBalanceTooLow   = errors.New("reserve balance amount is too low")
+
+	tenThousands = decimal.NewFromInt(10000)
+)
+
+type BalancedReservesOpts struct {
```

Comment:
why do we need formulas? isn't it enough to translate the strategy type in accordance with the changes made from v0.x to v1.x without balanced/pluggable implementations?

Reply:
I dont get how we fetch Market from v091 Db, having Strategy field as interface, unless we add implementation for both Balanced and Pluggable?

Reply:
Ii saw that the v0 domain stores the strategy as [`marketmaking.MarketStrategy`](https://github.com/tdex-network/tdex-daemon/blob/master/pkg/marketmaking/market_making.go#L8-L11) so I understood why we need the impls (because of `restoreStrategy`).

since we are not really interested in the formula impl, we could make changes to `NewBalancedStrategy` and `NewPluggableStrategy` factories defined here and just set to formula to `nil` for simplicity.

The best thing would be isolating this so that we can directly reuse pkg/marketmaking from tdex-daemon@v0.9.1. We'll discuss about this offline